interfaces/tests/file_int/file_int.c

```diff
@@ -0,0 +1,427 @@
+//Copyright(c) Microsoft.All rights reserved.
+//Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#endif
+
+
+#include "testrunnerswitcher.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+#include "azure_c_pal/sync.h"
+#include "azure_c_pal/file.h"
+#include "azure_c_pal/interlocked.h"
+
+
+TEST_DEFINE_ENUM_TYPE(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_RESULT)
+TEST_DEFINE_ENUM_TYPE(FILE_READ_ASYNC_RESULT, FILE_READ_ASYNC_RESULT)
+
+typedef struct WRITE_COMPLETE_CONTEXT_TAG
+{
+    int32_t pre_callback_value;
+    volatile int32_t value;

```

Comment:
>volatile [](start = 4, length = 8)

`volatile_atomic`? Here and everywhere

Reply:
done


---
In reply to: [456711837](https://github.com/Azure/azure-c-pal/pull/10#discussion_r456711837) [](ancestors = 456711837)